packages/central-server/src/tests/apiV2/helpers.test.js

```diff
@@ -0,0 +1,189 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import { expect } from 'chai';
+import { ValidationError } from '@tupaia/utils';
+import { getQueryOptionsForColumns } from '../../apiV2/GETHandler/helpers';
+
+const baseRecordType1 = 'survey_response';
+const baseRecordType2 = 'project';
+const columnNames1 = ['survey_response.id', 'survey.name'];
+const columnNames2 = ['survey_response.id', 'survey.name', 'user_account.first_name'];
+const columnNames3 = ['survey_response.id', 'country.name'];
+const columnNames4 = ['survey_response.id', 'country.name', 'disaster.type'];
+const columnNames5 = ['_survey_response.id', 'survey.name'];
+const columnNames6 = ['project.id', 'entity.id'];
+const customJoinConditions1 = {};
+const customJoinConditions2 = {
+  country: {
+    through: 'entity',
+    nearTableKey: 'entity.country_code',
+    farTableKey: 'country.code',
+  },
+};
+const customJoinConditions3 = {
+  country: {
+    through: 'entity',
+    nearTableKey: 'entity.country_code',
+    farTableKey: 'country.code',
+  },
+  disaster: {
+    through: 'country',
+    nearTableKey: 'country.code',
+    farTableKey: 'disaster.countryCode',
+  },
+};
+const customJoinConditions4 = {
+  country: {
+    through: 'entity',
+    nearTableKey: 'country.code',
+    farTableKey: 'entity.country_code',
+  },
+};
+const customJoinConditions5 = {
+  entity: ['entity.id', 'project.entity_id'],
+};
+const joinType = null;
+const err = new ValidationError(
+  'Error: No columns start with "_", and conjunction operators are reserved for internal use only',
+);
+const err2 = new ValidationError(
+  'nearTableKey must refer to a column on the table you wish to join through',
+);
+const err3 = new ValidationError('Incorrect format for customJoinConditions: entity');
```

Comment:
Rather than constructing the full exact error, you can use jest's `toThrow` and do a partial match on the error text